services/hackage/hackage-deps.service.js

```diff
@@ -22,62 +14,48 @@ module.exports = class HackageDeps extends LegacyService {
     }
   }
 
+  static get defaultBadgeData() {
+    return { label: 'dependencies' }
+  }
+
   static get examples() {
     return [
       {
         title: 'Hackage-Deps',
         namedParams: { packageName: 'lens' },
-        staticPreview: {
-          label: 'dependencies',
-          message: 'up to date',
-          color: 'brightgreen',
-        },
+        staticPreview: this.render({ message: 'up to date' }),
       },
     ]
   }
 
-  static registerLegacyRouteHandler({ camp, cache }) {
-    camp.route(
-      /^\/hackage-deps\/v\/(.*)\.(svg|png|gif|jpg|json)$/,
-      cache((data, match, sendBadge, request) => {
-        const repo = match[1] // eg, `lens`.
-        const format = match[2]
-        const reverseUrl = `http://packdeps.haskellers.com/licenses/${repo}`
-        const feedUrl = `http://packdeps.haskellers.com/feed/${repo}`
-        const badgeData = getBadgeData('dependencies', data)
+  static render({ message }) {
```

Comment:
Would passing a boolean `isUpToDate` (or `isOutdated`) keep this better encapsulated?